### PR TITLE
Spring2024 tokenomics rework drafts2

### PR DESCRIPTION
**Pull request by Valdorff**

### CODE REVIEW

File: RPIPs/RPIP-42.md [hunk 1]
```diff
@@ -23,18 +23,22 @@ This work is based on prior work; a copy can be found [here](../assets/rpip-42/b
...
 - The protocol SHALL be able to penalize stake at the node level
-- Node Operators with <`base_mp_num` base minipools SHALL be able to make a base minpool
-  - A base minipool SHALL have a `base_mp_bond` bond from the Node Operator
-- Node Operators with >=`base_mp_num` base minipools SHALL be able to make a satellite minipool
-  - A satellite minipool SHALL have a `satellite_mp_bond` bond from the Node Operator
-- If a Node has satellite minipools active while that Node has <`base_mp_num` base minipools active, funds SHALL NOT be withdrawable by the Node Operator
-  - It is NOT RECOMMENDED to exit base minipools unless all satellite minipools have been exited
-  - A user MAY exit satellite minipools to allow for fund withdrawal 
-  - The protocol MAY or MAY NOT provide a path to upgrade a satellite minipool to a base minipool (to allow for fund withdrawal)
+- Minipool deposits SHALL be disabled
```
Comment (pieter-astra) [inline RPIPs/RPIP-42.md:26]: This confused me at first read (thought it was an incomplete reference to a node being penalized.)
Could make it more explicit this is about old minipools (non-megapool validators.) 
(Why is it necessary to disable these immediately?)

File: RPIPs/RPIP-49.md [hunk 2]
```diff
@@ -15,39 +15,54 @@ This Info RPIP exists to explain an overall tokenomics proposal, the RPIPs it ha
...
-We'd love to get team feedback here, especially on ballpark timelines.
+In order to avoid messy conditionals in the rest of the RPIPs, we will be operating with "Scenario C" as the best guess.
 
 ### Scenario A (1 release):
-- Release 1: Components 1-7 at once.
+- Release 1: Components 1-6 at once.
 
 ### Scenario B (3 releases):
-- Release 1: Megapools, Remove operator-gated upgrades, Lower minipool requirements
+- Release 1: Megapools, Forced delegaet upgrades
```
Comment (pieter-astra) [inline RPIPs/RPIP-49.md:54]: typo in delegate a couple of times

File: RPIPs/RPIP-47.md [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
 
 ## Abstract
 Currently, Node Operators can choose to upgrade (or not) for their minipool delegate (the contract at the Ethereum withdrawal address that governs how funds are disbursed, among other things). This (a) means the protocol can only effect change in ways that are in the interest of individual NOs, and (b) makes it more challenging to design the protocol as it must remain backwards compatible with every past minipool delegate.
 
-This proposal suggests removing that choice in the future. Instead, a time lock will provide time for users to initiate withdrawals in a case where they are unwilling to operate under a new incoming set of rules.
+This proposal suggests limiting that upgrade choice in the future. Users will be able to opt in to upgrade or use an "old" delegate for a defined period after a newer version. However, once the defined period has passed, they will no longer use the old delegate.
 
 ## Specification
-- Contracts moving forward SHALL NOT require Node Operators to individually opt in to them
+- Megapool delegate contracts SHALL have an expiration block
```
Comment (pieter-astra) [inline RPIPs/RPIP-47.md:20]: Could clarify this is about an on-chain block number, not e.g. a 'block on expiration'

File: RPIPs/RPIP-49.md [hunk 4]
```diff
@@ -15,39 +15,54 @@ This Info RPIP exists to explain an overall tokenomics proposal, the RPIPs it ha
...
+3. [RPIP-44: Forced exits](RPIP-44.md) - this is needed for the bond curve changes, to enable node level penalties to be effective.
+4. [RPIP-45: RPL Burn](RPIP-45.md) - this is the primary value capture mechanism being proposed.
+5. [RPIP-46: UVC](RPIP-46.md) - this is the mechanism that allows for splitting revenue between various targets (including voters to keep our governance robust). Note that it also has some thinking about how to find an appropriate share for NOs, as well as some future-looking settings that are not intended to be used immediately.
+6. [RPIP-47: Forced delegate upgrades](RPIP-47.md) - this is desirable eventually. Since it reduces the number of things that need to be tested going forward, the thought is we should include it as quickly as possible.
 
-1 requires 2 and 3.
-4 and 5 require each other.
+## Topics that need feedback, discussion, or fleshing out
+- [RPL Burn](RPIP-45.md) vs [RPL Buy & LP](RPIP-50.md)
+- 3-transaction deposits for greater queue efficiency
```
Comment (pieter-astra) [inline RPIPs/RPIP-49.md:27]: What are 3-transaction deposits? Can you summarize or add a link to discussion of the concept?

File: RPIPs/RPIP-49.md [hunk 5]
```diff
@@ -15,39 +15,54 @@ This Info RPIP exists to explain an overall tokenomics proposal, the RPIPs it ha
...
 - Release 2: RPL burn, UVC
   - New RPL value capture comes online
   - Revenue steering controls come online
 - Release 3: Bond curves, Forced exits
   - Dramatically improve capital efficiency 
 
-### Scenario C (2 releases):
-- Release 1: Megapools, Remove operator-gated upgrades, Lower minipool requirements, RPL burn, UVC
+### Scenario C (2 releases; currently assumed for all RPIPs):
+- Release 1: Megapools, Forced delegaet upgrades, RPL burn, UVC
```
Comment (stevenpack) [inline RPIPs/RPIP-49.md:65]: typo: delegate

File: RPIPs/RPIP-42.md [hunk 6]
```diff
@@ -23,18 +23,22 @@ This work is based on prior work; a copy can be found [here](../assets/rpip-42/b
...
+- Node Operators SHALL NOT be required to stake RPL in order to create validators within their megapool
+- Node Operators with <`base_num` base validators SHALL be able to make a base validator within their megapool
+  - A base validator SHALL have a `base_bond` bond from the Node Operator
+  - If there is a node operator queue, base validators SHALL be given priority over satellite validators
+- Node Operators with >=`base_num` base validators SHALL be able to make a satellite validator
+  - A satellite validator SHALL have a `satellite_bond` bond from the Node Operator
```
Comment (stevenpack) [inline RPIPs/RPIP-42.md:32]: Suggest defining satellite validator. I don't know what it means and googling did not surface in the RocketPool docs

File: RPIPs/RPIP-49.md [hunk 7]
```diff
@@ -15,39 +15,54 @@ This Info RPIP exists to explain an overall tokenomics proposal, the RPIPs it ha
...
-6. [Remove operator-gated upgrades](RPIP-47.md) - this is desirable eventually. Since it reduces the number of things that need to be tested going forward, the thought is we should include it as quickly as possible.
-7. [Lower minipool requirements](RPIP-48.md) - this (a) enables a new size of minipool with 4 ETH bond from the NO and (b) removes the RPL staking requirement to create minipools. If not released alongside UVC, rETH commisison is changed to make these new pools only slightly more productive than the current 8-ETH minipools.
+1. [RPIP-42: Bond curves](RPIP-42.md) - this is perhaps the most critical component, as it unlocks dramatically higher capital efficiency. Note that this RPIP also includes (a) the ability to penalize at the node level and (b) the removal of the RPL staking requirement to create minipools.
+2. [RPIP-43: Megapools](RPIP-43.md) - this is needed for the bond curve changes, otherwise gas would be prohibitive gas for, eg, 1.5-ETH bond minipools.
```
Comment (stevenpack) [inline RPIPs/RPIP-49.md:19]: Is there any calculations for this anywhere? My node shows .2 ETH gas. How long would it take to pay off that with the increased commission? It probably makes sense to wait for Megapools anyway, because you'd duplicate a bunch of the logic I imagine, but if say Megapool was >= 1 year away and bonding curves could be <= 3 months away, may it would be worth it?

Comment (knoshua) [inline RPIPs/RPIP-49.md:19]: ...otherwise gas would be prohibitive ~~gas~~ for

File: RPIPs/RPIP-49.md [hunk 8]
```diff
@@ -15,38 +15,54 @@ This Info RPIP exists to explain an overall tokenomics proposal, the RPIPs it ha
...
   - Revenue steering controls come online
 - Release 3: Bond curves, Forced exits
   - Dramatically improve capital efficiency 
 
-### Scenario C (2 releases):
-- Release 1: Megapools, Remove operator-gated upgrades, RPL burn, UVC
+### Scenario C (2 releases; currently assumed for all RPIPs):
+- Release 1: Megapools, Forced delegaet upgrades, RPL burn, UVC
   - Allows ETH-only participants
   - Allows NOs to migrate to 4-ETH minipools for greater efficiency
```
Comment (orangesamus) [inline RPIPs/RPIP-49.md:67]: Change language to match Scenario B:
- Allows NOs to move to Megapools for greater efficiency
- Allow 4-ETH validators within Megapools

Avoids confusion compared to "migrating to 4-ETH minipools"

File: RPIPs/RPIP-42.md (hunk 1 above)
Comment (Valdorff) [reply RPIPs/RPIP-42.md:26]: I'll say "legacy minipool" and make it a softer SHOULD. I think they will be categorically worse and see no reason to give users an extra way to shoot themselves in the foot.

File: RPIPs/RPIP-49.md [hunk 9]
```diff
@@ -56,41 +72,15 @@ We'd love to get team feedback here, especially on ballpark timelines.
   - Dramatically improve capital efficiency 
 
 ### Scenario D (2 releases):
-- Release 1: Megapools, Remove operator-gated upgrades
+- Release 1: Megapools, Forced delegaet upgrades
   - Allows ETH-only participants
   - Allows NOs to migrate to 4-ETH minipools for greater efficiency
```
Comment (orangesamus) [inline RPIPs/RPIP-49.md:77]: Change language to match Scenario B:

    Allows NOs to move to Megapools for greater efficiency
    Allow 4-ETH validators within Megapools

Avoids confusion compared to "migrating to 4-ETH minipools"

File: RPIPs/RPIP-42.md (hunk 6 above)
Comment (Valdorff) [reply RPIPs/RPIP-42.md:32]: This is the document that defines them. It defines their size, when they can be made etc

File: RPIPs/RPIP-49.md (hunk 7 above)
Comment (Valdorff) [reply RPIPs/RPIP-49.md:19]: Bonding curves have more dependencies than Megapools, so I think its moot

**Review by orangesamus — COMMENTED** [review]
Update language for megapools

File: RPIPs/RPIP-42.md [hunk 10]
```diff
@@ -22,19 +22,35 @@ This proposal dramatically increases the LTV used in the protocol (loan to value
...
-- If a Node has satellite minipools active while that Node has <`base_mp_num` base minipools active, funds SHALL NOT be withdrawable by the Node Operator
-  - It is NOT RECOMMENDED to exit base minipools unless all satellite minipools have been exited
-  - A user MAY exit satellite minipools to allow for fund withdrawal 
-  - The protocol MAY or MAY NOT provide a path to upgrade a satellite minipool to a base minipool (to allow for fund withdrawal)
+- The oDAO SHALL be able to penalize stake at the node level when a [Penalizable offense](#penalizable-offenses) is committed
+- Legacy minipool deposits SHOULD be disabled
+- Node Operators SHALL NOT be required to stake RPL in order to create validators within their megapool
+- When Node Operators create validators:
+  - If fewer than `base_num` validators already exist, the required `user_deposit` is `base_bond` per validator.
+  - If at least `base_num` validators already exist, the required `user_deposit` is `reduced_bond` per validator.
```
Comment (htimsk) [inline RPIPs/RPIP-42.md:30]: Was this meant to be the same? Should there not be a distinction between < base = 4 ETH and >= base at 1.5 ETH. (Or the ETH number that we specify). 
Also the way this is written base = 3 in the 4+4+1.5 model.

File: RPIPs/RPIP-44.md [hunk 11]
```diff
@@ -12,22 +12,45 @@ requires: eip-7002
...
+- `keeper_reward` is set to ensure that users are incentivized to call this function when possible
+- `expected_credit_from_exit` has a known potential shortfall in cases of correlated slashing; that said, upon the completion of the exits, the numbers will be correctly updated. This means there may be a second round of exits in such a case.
+
+## Risk Analysis
+
+### Forced Exits During Inactivity Leaking
+
+The Ethereum staking formula includes additional penalties for offline validators when large portions of the network are offline. This state of accelerated punishment is called ["inactivity leaking"](https://ethereum.org/en/developers/docs/consensus-mechanisms/pos/rewards-and-penalties/#inactivity-leak). With the RP protocol forcing exits for validators under a 31.9 ETH balance, validators who are offline during an inactivity leaking scenario may be exited even if the validator is only offline for a brief period. In some extreme scenarios, a small number of missed attestations could cause a validator to be ejected from the network. This may be necessary, however, in order to protect the value of rETH during an unusual network state for Ethereum.
```
Comment (htimsk) [inline RPIPs/RPIP-44.md:46]: 31.9 ETH does not align with the `exit_deficit` value of 0.5 ETH.

**Review by htimsk — APPROVED** [review]
Made a few comments for consideration. Finished review as approve.

File: RPIPs/RPIP-44.md (hunk 11 above)
Comment (Valdorff) [reply RPIPs/RPIP-44.md:46]: badly written in general -- merged a per-validator and a per-node thing here

File: RPIPs/RPIP-42.md [hunk 12]
```diff
@@ -22,19 +22,35 @@ This proposal dramatically increases the LTV used in the protocol (loan to value
...
+- When Node Operators remove validators:
+  - If greater than `base_num` validators exist prior to removing, the Node Operator share before penalties is `reduced_bond`.
+  - If at most `base_num` validators exist prior to removing, the Node Operator share before penalties is `base_bond`.
+- When a validator is added with a `base_bond` deposit, it SHALL receive priority treatment for any queues vis-à-vis validators added with a `reduced_bond` deposit.
+- Bulk validator creation / removal functions SHALL behave the same as multiple individual transactions.
 - The initial settings SHALL be:
-  - `base_mp_num`: 2
-  - `base_mp_bond`: 4 ETH
-  - `satellite_mp_bond`: 1.5 ETH
+  - `base_num`: 2
```
[the remark this answers is not in the document]
Comment (htimsk) [reply RPIPs/RPIP-42.md:37]: Val's clarification :
Seems like it works just fine right now? Let's use base_num=2, base_bond=4, reduced_bond=1.5 and https://github.com/Valdorff/RPIPs/blob/spring2024-tokenomics-rework-drafts2/RPIPs/RPIP-42.md#specification.

Now let's make some deposits:
You have 0 validators, which is <2, so your user_deposit is 4
You have 1 validator, which is <2, so your user_deposit is 4
You have 2 validators, which is >=2, so your user_deposit is 1.5
You have n validators, which is >=2, so your user_deposit is 1.5

File: RPIPs/RPIP-42.md (hunk 10 above)
Comment (htimsk) [reply RPIPs/RPIP-42.md:30]: This was Val's clarification:
Seems like it works just fine right now? Let's use base_num=2, base_bond=4, reduced_bond=1.5 and https://github.com/Valdorff/RPIPs/blob/spring2024-tokenomics-rework-drafts2/RPIPs/RPIP-42.md#specification.

Now let's make some deposits:
You have 0 validators, which is <2, so your user_deposit is 4
You have 1 validator, which is <2, so your user_deposit is 4
You have 2 validators, which is >=2, so your user_deposit is 1.5
You have n validators, which is >=2, so your user_deposit is 1.5
